packages/oid4vci-demo-frontend/src/components/Form/index.tsx

```diff
@@ -0,0 +1,103 @@
+import React, {FC, ReactElement, ReactNode, useState} from 'react';
+import {SSICheckbox} from '@sphereon/ui-components.ssi-react';
+import {useTranslation} from 'react-i18next';
+import {DataFormElement, DataFormRow} from '../../ecosystem-config';
+import {generateRandomIBAN} from '../../utils/iban';
+import {transformFormConfigToEmptyObject} from '../../utils/ObjectUtils';
+import InputField from '../InputField';
+import {FormData, FormFieldValue} from '../../types';
+import style from './index.module.css';
+
+type Props = {
+    form: DataFormRow[]
+    onChange?: (formData: FormData) => Promise<void>
+}
+
+function getInitialState(form: DataFormRow[] | undefined): FormData {
+    if (!form) {
+        return {
+            firstName: '',
+            lastName: '',
+            emailAddress: ''
+        }
+    }
+    return transformFormConfigToEmptyObject(form)
+}
+
+const evaluateDefaultValue = (field: DataFormElement, payload: FormData): FormFieldValue => {
+    const payloadValue = payload[field.key]
+    if (payloadValue) {
+        return payloadValue
+    }
+
+    let defaultValue: FormFieldValue = field.defaultValue ?? ''
+    if (defaultValue === '*RANDOM8') { // TODO this is for a demo, create something more sophisticated later
+        defaultValue = Math.floor(Math.random() * 89999999 + 10000000)
+    } else if (defaultValue === '*RANDOM-IBAN') { // TODO this is for a demo, create something more sophisticated later
+        defaultValue = generateRandomIBAN()
+    }
+    payload[field.key] = `${defaultValue}`
+    return defaultValue
+}
+
+const Form: FC<Props> = (props: Props): ReactElement => {
+    const { form, onChange } = props
+    const {t} = useTranslation()
+    const [formData, setFormData] = useState<FormData>(getInitialState(form))
+
+    const onChangeValue = async (value: FormFieldValue, key: string): Promise<void> => {
+        const data = {...formData, [key]: value}
+        setFormData(data)
+        if (onChange) {
+            await onChange(data)
+        }
+    }
+
+    const getFormFields = (): Array<ReactElement> => {
+        return form.map((row: DataFormRow) =>
+                <div className={style.formContentContainer}>
+                    {row.map((field: DataFormElement): ReactNode => {
+                        const defaultValue: FormFieldValue = evaluateDefaultValue(field, formData)
```

Comment:
Not nicer to have either a buildField and maybe buildRow functions? It's getting nested quite deep like this

Reply:
ye, that is fair enough. i cleaned it up a bit